src/test/java/com/conveyal/datatools/manager/jobs/JobChainTest.java

```diff
@@ -0,0 +1,63 @@
+package com.conveyal.datatools.manager.jobs;
+
+import com.conveyal.datatools.DatatoolsTest;
+import com.conveyal.datatools.TestUtils;
+import com.conveyal.datatools.manager.auth.Auth0UserProfile;
+import com.conveyal.datatools.manager.models.FeedRetrievalMethod;
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test class for checking job chaining functionality.
+ */
+class JobChainTest extends DatatoolsTest {
+    private static final Logger LOG = LoggerFactory.getLogger(JobChainTest.class);
+    private static final Auth0UserProfile user = Auth0UserProfile.createTestAdminUser();
+    private static FeedSource mockFeedSource;
+    private static Project project;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        DatatoolsTest.setUp();
+        LOG.info("{} setup", JobChainTest.class.getSimpleName());
+
+        // Create a project, feed sources to pass to jobs.
+        project = new Project();
+        project.name = String.format("Test %s", new Date());
+        Persistence.projects.create(project);
+        mockFeedSource = new FeedSource("Mock Feed Source", project.id, FeedRetrievalMethod.MANUALLY_UPLOADED);
+        Persistence.feedSources.create(mockFeedSource);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        mockFeedSource.delete();
+        project.delete();
+    }
+
+    @Test
+    void shouldPopulateParentJobIdAndTypeInSubjobs() {
+        FeedVersion f = TestUtils.createMockFeedVersion(mockFeedSource.id);
+
+        // Define a simple job chain.
+        // (The data for these jobs is arbitrary, the jobs are not actually executed.)
+        ProcessSingleFeedJob parentJob = new ProcessSingleFeedJob(f, user, true);
+        FetchSingleFeedJob subJob = new FetchSingleFeedJob(mockFeedSource, user, true);
+        parentJob.addNextJob(subJob);
+
+        assertEquals(parentJob.jobId, subJob.parentJobId);
+        assertEquals(parentJob.type, subJob.parentJobType);
```

Comment:
This is good, but I was thinking that it might be more appropriate to create a StatusControllerTest class and verify the JSON response from the API contains these fields with some snapshots (given that the main concern is that these fields make their way back through to the UI.

Reply:
Good point. Changed in https://github.com/ibi-group/datatools-server/pull/406/commits/5c99775627b6b504811a87fb2c548ee2e948cf2e.